src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.SingleStatementCodeGenerator.cs

```diff
@@ -28,38 +25,15 @@ public sealed class SingleStatementCodeGenerator(
                 CSharpCodeGenerationOptions options,
                 bool localFunction) : CSharpCodeGenerator(insertionPoint, selectionResult, analyzerResult, options, localFunction)
             {
-                public static bool IsExtractMethodOnSingleStatement(SelectionResult code)
-                {
-                    var result = (CSharpSelectionResult)code;
-                    var firstStatement = result.GetFirstStatement();
-                    var lastStatement = result.GetLastStatement();
-
-                    return firstStatement == lastStatement || firstStatement.Span.Contains(lastStatement.Span);
-                }
```

Comment:
this logic moved from the generator to the 'selection result' type (where it makes much more sense) allowing certain checks to be front loaded prior to the generation step.